### PR TITLE
[CALCITE-6370] AS operator problems with USING clause

### PR DESCRIPTION
**Pull request by normanj-bitquill**
* Detect when a nested AS call is inserted and prevent it

### CODE REVIEW

**Review by caicancai — APPROVED** [review]
LGTM

**Review by YiwenWu — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/apache/calcite/sql/util/SqlShuttle.java [hunk 1]
```diff
@@ -135,6 +136,14 @@ public CallCopyingArgHandler(SqlCall call, boolean alwaysCopy) {
       if (newOperand != operand) {
         update = true;
       }
+
+      // Prevent nesting AS calls
+      if (i == 0 && expr instanceof SqlCall
```
[the remark this answers is not in the document]
Comment (normanj-bitquill) [reply core/src/main/java/org/apache/calcite/sql/util/SqlShuttle.java]: @YiwenWu  I have reworked this to much more closely match what you describe above.

**Review by YiwenWu — APPROVED** [review]

File: core/src/main/java/org/apache/calcite/sql/util/SqlShuttle.java [hunk 2]
```diff
@@ -135,6 +135,7 @@ public CallCopyingArgHandler(SqlCall call, boolean alwaysCopy) {
       if (newOperand != operand) {
         update = true;
       }
+
```
Comment (NobiGo) [inline core/src/main/java/org/apache/calcite/sql/util/SqlShuttle.java:138]: Please remove the blank lines.

**Review by NobiGo — APPROVED** [review]

File: core/src/main/java/org/apache/calcite/sql/util/SqlShuttle.java (hunk 2 above)
Comment (normanj-bitquill) [reply core/src/main/java/org/apache/calcite/sql/util/SqlShuttle.java]: Updated.

**Comment by mihaibudiu** [comment]
this seems to be approved, can you squash the commits?

**Comment by normanj-bitquill** [comment]
@mihaibudiu The changes have been squashed.

**Pull request merged**